src/internet_identity/src/main.rs

```diff
@@ -728,20 +765,50 @@ mod v2_api {
         Ok(())
     }
 
+    impl From<IdentityUpdateError> for IdentityMetadataReplaceError {
+        fn from(value: IdentityUpdateError) -> Self {
+            let storage_err = match value {
+                IdentityUpdateError::Unauthorized => {
+                    return IdentityMetadataReplaceError::Unauthorized
+                }
+                IdentityUpdateError::StorageError(_, storage_err) => storage_err,
+            };
+
+            match storage_err {
+                StorageError::EntrySizeLimitExceeded(_) => {
+                    IdentityMetadataReplaceError::StorageSpaceExceeded
```

Comment:
nit: also here, maybe we should keep the extra information in `StorageSpaceExceeded` (effective size, limit) available for debugging?